src/util.rs

```diff
@@ -55,3 +55,332 @@ impl Row for Index {
 pub fn offsets(bit: Index) -> (usize, usize, usize) {
     (bit.offset(SHIFT1), bit.offset(SHIFT2), bit.offset(SHIFT3))
 }
+
+/// Finds the highest bit that splits set bits of the `usize`
+/// to half (rounding up).
+///
+/// Returns `None` if the `usize` has only one or zero set bits.
+///
+/// # Examples
+/// ````rust,ignore
+/// use hibitset::util::average_ones;
+///
+/// assert_eq!(Some(3), average_ones(0b10110));
+/// assert_eq!(Some(6), average_ones(0b100010));
+/// assert_eq!(None, average_ones(0));
+/// assert_eq!(None, average_ones(1));
+/// ````
+// TODO: Can 64/32 bit variants be merged to one implementation?
+// Seems that this would need integer generics to do.
+pub fn average_ones(n: usize) -> Option<usize> {
+    #[cfg(target_pointer_width = "64")]
+    let average = average_ones_u64(n as u64).map(|n| n as usize);
+
+    #[cfg(target_pointer_width = "32")]
+    let average = average_ones_u32(n as u32).map(|n| n as usize);
+
+    average
+}
+
+#[allow(dead_code)]
+fn average_ones_u32(n: u32) -> Option<u32> {
```

Comment:
why is this `dead_code` as opposed to straight be gated by `target_pointer_width`?

Reply:
Because it's used by tests. We want to run the tests for both variants.

It could be possible gate it with tests too though...

Reply:
please, yes, let's do `cfg(test)`